### DISCUSSION TITLE
Next.js Supabase - Google auth will not redirect me to the provided url.

### DISCUSSION THREAD

**Discussion by teodor1608** [supabase/supabase#23380] [Questions]
Hi, I am using Supa for authentication however the google authentication flow would NOT redirect me to my auth callback route whatsoever. This is supa config
![image](https://github.com/supabase/supabase/assets/36604281/764c316b-878a-4ef9-a0b6-95a7f7bc4864)
This is google
![image](https://github.com/supabase/supabase/assets/36604281/ae0d858b-109b-4a91-958b-e3dd307f7bac)
And the code used to authenticate.
![image](https://github.com/supabase/supabase/assets/36604281/5320be6c-d48e-4b2d-98e0-db98c7506538)
Outcome is that i always get redirected to http://localhost:3000/?code=***

Please help!

**Comment by awalias** [comment]
what is hosted at `http://localhost:3000/auth/callback`? Could that be forwarding you back to `http://localhost:3000/?code=`?

**Reply by teodor1608** [reply]
This is the code. I am almost certain it's not it because when I add /auth/callback it just logs me in as it should.
Thanks a lot for the reply.

import { NextResponse } from "next/server";
import { cookies } from 'next/headers'
import { createServerClient } from "@supabase/ssr";

export async function GET(request) {
    const { searchParams, origin } = new URL(request.url);
    const cookieStore = cookies();

    const supabase = createServerClient(
        process.env.NEXT_PUBLIC_SUPABASE_URL,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY,
        {
            cookies: {
                get(name){
                    return cookieStore.get(name)?.value
                },
                set(name, value, options){
                    cookieStore.set({name, value, ...options})
                },
                remove(name, options){
                    cookieStore.set({name, value: '', ...options})
                }
            }
        }
    )
  
    const code = searchParams.get("code");

    // if "next" is in param, use it in the redirect URL
    const next = searchParams.get("next") ?? "/";

    if (code) {

        const { error } = await supabase.auth.exchangeCodeForSession(code);

        if (!error) {
        return NextResponse.redirect(`${origin}${next}`);
        }
    }

    // TODO: Create this page
    // return the user to an error page with instructions
    return NextResponse.redirect(`${origin}/auth/auth-error`);
}
